File: backend/passatempo/src/main/java/com/luisn/passatempo/domain/Locacao.java
package com.luisn.passatempo.domain;

import jakarta.persistence.*;
import lombok.Data;
import java.util.Date;

@Data
@Entity
public class Locacao {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Temporal(TemporalType.DATE)
    private Date dt_locacao;

    @Temporal(TemporalType.DATE)
    private Date dt_devolucaoPrevista;

    @Temporal(TemporalType.DATE)
    private Date dtDevolucaoEfetiva;

    @Column
    private float valor;

    @Column
    private float multa;

    @ManyToOne
    @JoinColumn(name = "id_item")
    private Item item;

    @ManyToOne
    @JoinColumn(name = "id_cliente")
    private Cliente cliente;

}

File: frontend/passatempo/src/app/models/locacao.ts
import { Cliente } from "./cliente";
import { Item } from "./item";

export class Locacao {
   constructor (
      public id?: string,
      public dt_locacao?: Date,
      public dt_devolucaoPrevista?: Date,
      public dtDevolucaoEfetiva?: Date,
      public valor?: number,
      public multa?: number,
      public item?: Item,
      public cliente?: Cliente
   ) { }
}